packages/api/src/shared/filter-map-utils.ts

```diff
@@ -1,3 +1,12 @@
+/**
+ * Used in .flatMap() to filter out undefined values.
+ *
+ * Example:
+ *
+ * const arr = [1, null, 3, undefined, 5];
+ * const filtered = arr.flatMap(filterTruthy);
+ * // filtered = [1, 3, 5]
```

Comment:
this kind of stuff is great ❤️ 

makes it way quicker to grok what's going on in the codebase